fs/inode/fs_files.c

```diff
@@ -111,7 +118,6 @@ static FAR struct file *files_fget_by_index(FAR struct filelist *list,
     }
 #endif
 
-  spin_unlock_irqrestore(&list->fl_lock, flags);
```

Comment:
too many unlock.... maybe we can keep this line

Reply:
there is a special logic  "filep = NULL;"
![image](https://github.com/user-attachments/assets/b96a0e4d-72d7-4c84-add4-4303d36fee63)